shipyard_blueprints/sqlserver/shipyard_sqlserver/sqlserver.py

```diff
@@ -20,19 +30,109 @@ def __init__(
         self.database = database
         self.port = port
         self.url_params = url_params
+        self._conn = None
         super().__init__(
             user, pwd, host=host, database=database, port=port, url_params=url_params
         )
 
+    @property
+    def conn(self):
+        if self._conn is None:
+            self.connect()
+        return self._conn
+
     def connect(self):
-        connection_string = f"DRIVER={{ODBC Driver 18 for SQL Server}};SERVER={self.host};PORT={self.port};DATABASE={self.database};UID={self.user};PWD={self.pwd};TrustServerCertificate=yes;"
-        return pyodbc.connect(connection_string)
+        """
+        Connects to the SQL Server database using the provided credentials.
+
+        Raises:
+            SqlServerConnectionError: If an error occurs while connecting to the database.
+
+        Returns:
+            None
+        """
+        try:
```

Comment:
This loses the current functionality of having the user explicitly set the connection url

Reply:
I actually need to remove that as an argument. It is not needed and the current blueprints do not even support it